gobblin-cluster/src/main/java/org/apache/gobblin/cluster/HelixRetriggeringJobCallable.java

```diff
@@ -196,22 +202,57 @@ private void deleteJobSpec() throws JobException {
    * the job launcher determines it is safe to stop.
    */
   private void runJobLauncherLoop() throws JobException {
+    String actualJobId;
+    Lock jobLock = locks.get(this.jobUri);
```

Comment:
It is to avoid race condition in checking if the job with the same name is running. this is a new thing this method is doing.
reference https://github.com/apache/incubator-gobblin/blob/master/gobblin-cluster/src/main/java/org/apache/gobblin/cluster/HelixRetriggeringJobCallable.java#L240